internal/exec/util/file.go

```diff
@@ -162,66 +161,133 @@ func (u Util) PerformFetch(f *FetchOp) error {
 		return err
 	}
 
-	// XXX(vc): Note that we assume to be operating on the file we just wrote, this is only guaranteed
-	// by using syscall.Fchown() and syscall.Fchmod()
+	if f.Append {
+		// Default to the appended file's owner for the uid and gid
+		defaultUid, defaultGid, mode := getFileOwnerAndMode(path)
+		uid, gid, err := u.GetNodeOwners(f.Node, defaultUid, defaultGid)
+		if err != nil {
+			return err
+		}
+		if f.Mode != nil {
+			mode = os.FileMode(*f.Mode)
+		}
 
-	// Ensure the ownership and mode are as requested (since WriteFile can be affected by sticky bit)
-	if err = os.Chown(tmp.Name(), f.Uid, f.Gid); err != nil {
-		return err
-	}
+		targetFile, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE, mode)
```

Comment:
It appears that this will write through a symlink.  That will have surprising results if the link target is an absolute path, or a relative path that crosses mountpoints.

Reply:
Added a check to ensure that the path being appended to is either empty or a regular file